Bottom: e959c3b0d67a3b16e5243fd69184937bf59c0b38
Top:    1d69d7e2ae62dc49d59d5a86f54323da85987358
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-16 10:06:33 -0500

Refresh of codebase-generator

---
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index 5b160a33..bc2acccb 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -6,7 +6,7 @@
 
 import os
 from buildbot.plugins import secrets, steps, util
-from urlparse import urlparse
+from buildbot.util import giturlparse
 
 os.environ['LONG_RUN_TIMEOUT'] = '14400'
 
@@ -151,17 +151,20 @@ def codebaseGenerator(chdict):
     This is used to decide upon the builder for a change and also to tie a fork
     repository back to its parent for merge requests.
     """
-    path = urlparse(chdict['revlink']).path.lstrip('/')
 
-    components = []
-    while path != "":
-        path, tail = os.path.split(path)
-        if tail != "":
-            components.append(tail)
-    components.reverse()
+    giturl = giturlparse(chdict['repository'])
+    return giturl.owner
 
-    codebase = os.path.join(*components[0:2])
-    return codebase
+    # path = urlparse(chdict['revlink']).path.lstrip('/')
+
+    # components = []
+    # while path != "":
+    #     path, tail = os.path.split(path)
+    #     if tail != "":
+    #         components.append(tail)
+    # components.reverse()
+    # codebase = os.path.join(*components[0:2])
+    # return codebase
 
 
 c['codebaseGenerator'] = codebaseGenerator
